### DISCUSSION TITLE
Pattern for CDK environments?

### DISCUSSION THREAD

**Discussion by automartin5000** [projen/projen#3188] [Q&A]
Is there a pattern for how to manage CDK environments? 

It seems like I'll want to create `awsEnvironments` as a required param in my custom project and then save it to a file somewhere. I was thinking about then storing it in package.json, but ran into [this](https://stackoverflow.com/a/53836076). Then I was thinking I'd use projen environment variables, but then that breaks the pattern of having the environment info committed and stored alongside the code. So then I was going to create a new managed file that's stored at the src root, but didn't immediately see how to create a new managed file and started wondering if there was some existing patterns here.

**Comment by hoegertn** [comment]
Hi, can you elaborate a bit on what you are trying to do?

A tool I am working on might help depending on what you want to achieve: https://github.com/taimos/projen-pipelines

**Comment by automartin5000** [comment]
Thanks @hoegertn. Yeah I think your [stages](https://github.com/taimos/projen-pipelines/blob/26c2b61ba09ba7b096bc13282de23f50cbd1240c/README.md?plain=1#L48) is kind of what I was going for with a slightly different implementation (external vars file with a loop in app.ts)

And actually the entire library does look like what I'm trying to build myself. What's the plan/timeline for that?

**Comment by hoegertn** [comment]
Gitlab and Github deployments are working and I am using it in several production projects.
The additional steps like test, security, etc are on the near-term roadmap. January should bring more features to the project.

Happy to have feedback on functionality and priority of the steps.

**Comment by automartin5000** [comment]
> Gitlab and Github deployments are working and I am using it in several production projects. The additional steps like test, security, etc are on the near-term roadmap. January should bring more features to the project.
> 
> Happy to have feedback on functionality and priority of the steps.

That all sounds great. I'll keep an eye on it and maybe test it out.

That said, I'm curious what other options people have explored. Not totally crazy about managing the app.ts file as I think there could be custom application logic in there.
